src/deposit/execute_deposit_utils.cairo

```diff
@@ -80,29 +104,390 @@ struct ExecuteDepositCache {
+            receiver: deposit.receiver,
+            ui_fee_receiver: deposit.ui_fee_receiver,
+            token_in: market.long_token,
+            token_out: market.short_token,
+            token_in_price: prices.long_token_price,
+            token_out_price: prices.short_token_price,
+            amount: cache.long_token_amount,
+            price_impact_usd: precision::mul_div(
+                i128_to_u128(cache.price_impact_usd),
+                cache.long_token_usd,
+                cache.long_token_usd + cache.short_token_usd
+            )
+        };
+
+        cache.received_market_tokens += _execute_deposit(@params, @_params);
+    }
+
+    if cache.short_token_amount > 0 {
+        let _params = _ExecuteDepositParams {
+            market: market,
+            account: deposit.account,
+            receiver: deposit.receiver,
+            ui_fee_receiver: deposit.ui_fee_receiver,
+            token_in: market.short_token,
+            token_out: market.long_token,
+            token_in_price: prices.short_token_price,
+            token_out_price: prices.long_token_price,
+            amount: cache.short_token_amount,
+            price_impact_usd: precision::mul_div(
+                i128_to_u128(cache.price_impact_usd),
+                cache.short_token_usd,
+                cache.long_token_usd + cache.short_token_usd
+            )
+        };
+
+        cache.received_market_tokens += _execute_deposit(@params, @_params);
+    }
+
+    if cache.received_market_tokens < deposit.min_market_tokens {
+        DepositError::MIN_MARKET_TOKENS(cache.received_market_tokens, deposit.min_market_tokens);
+    }
+
+    market_utils::validate_market_token_balance(params.data_store, market);
+
+    (params.event_emitter)
+        .emit_deposit_executed(
+            params.key,
+            cache.long_token_amount,
+            cache.short_token_amount,
+            cache.received_market_tokens,
+        );
+
+    let event_data = EventLogData {
+        uint_items: UintItems { items: array![], array_items: array![] }, cant_be_empty: 0
+    };
+    event_data.uint_items.init_items(1);
+    event_data.uint_items.set_item(0, 'received_market_tokens', cache.received_market_tokens);
+    after_deposit_execution(params.key, deposit, event_data, params.event_emitter);
+
+    pay_execution_fee_deposit(
+        params.data_store,
+        params.event_emitter,
+        params.deposit_vault,
+        deposit.execution_fee,
+        params.starting_gas,
+        params.keeper,
+        deposit.account,
+    );
 }
 
 /// Executes a deposit.
 /// # Arguments
-/// * `params` - ExecuteDepositParams.
-/// * `_params` - _ExecuteDepositParams.
+/// * `params` - @ExecuteDepositParams.
+/// * `_params` - @_ExecuteDepositParams.
 #[inline(always)]
-fn _execute_deposit(params: ExecuteDepositParams, _params: _ExecuteDepositParams) -> u128 {
-    //TODO
-    0
+fn _execute_deposit(params: @ExecuteDepositParams, _params: @_ExecuteDepositParams) -> u128 {
```

Comment:
Change the name of the function please we decided not to use the _ at the beginning of the function name

Reply:
@Sk8erboi84 sounds good, I didn't change anything in the function signature and just worked from there (it's also the same in gmx). What should I change it to?

Reply:
Change it to a coherent function name